android/src/main/java/com/oktareactnative/HttpClientImpl.java

```diff
@@ -12,166 +12,161 @@
+    protected Request buildRequest(Uri uri, ConnectionParameters param) {
+        if (sOkHttpClient == null) {
+            sOkHttpClient = new OkHttpClient.Builder()
+                    .connectTimeout(connectTimeoutMs, TimeUnit.MILLISECONDS)
+                    .readTimeout(readTimeoutMs, TimeUnit.MILLISECONDS)
+                    .build();
         }
-
-        conn.setConnectTimeout(connectTimeoutMs);
-        conn.setReadTimeout(readTimeoutMs);
-        conn.setInstanceFollowRedirects(false);
-
-        Map<String, String> requestProperties = params.requestProperties();
-        String userAgent = getUserAgent();
-        requestProperties.put(USER_AGENT, userAgent);
-        for (String property : requestProperties.keySet()) {
-            conn.setRequestProperty(property, requestProperties.get(property));
+        Request.Builder requestBuilder = new Request.Builder().url(uri.toString());
+        requestBuilder.addHeader(USER_AGENT, getUserAgent());
+        for (Map.Entry<String, String> headerEntry : param.requestProperties().entrySet()) {
+            String key = headerEntry.getKey();
+            requestBuilder.addHeader(key, headerEntry.getValue());
         }
-
-        ConnectionParameters.RequestMethod requestMethod = params.requestMethod();
-        Map<String, String> postParameters = params.postParameters();
-        conn.setRequestMethod(requestMethod.name());
-        if (requestMethod == ConnectionParameters.RequestMethod.GET) {
-            conn.setDoInput(true);
-        } else if (requestMethod == ConnectionParameters.RequestMethod.POST) {
-            conn.setDoOutput(true);
-            if (postParameters != null && !postParameters.isEmpty()) {
-                DataOutputStream out = new DataOutputStream(conn.getOutputStream());
-                out.write(params.getEncodedPostParameters());
-                out.close();
+        if (param.requestMethod() == ConnectionParameters.RequestMethod.GET) {
+            requestBuilder = requestBuilder.get();
+        } else {
+            Map<String, String> postParameters = param.postParameters();
+            if (postParameters != null) {
+                FormBody.Builder formBuilder = new FormBody.Builder();
+                for (Map.Entry<String, String> postEntry : postParameters.entrySet()) {
+                    String key = postEntry.getKey();
+                    formBuilder.add(key, postEntry.getValue());
+                }
+                RequestBody formBody = formBuilder.build();
+                requestBuilder.post(formBody);
+            } else {
+                requestBuilder.post(RequestBody.create(null, ""));
             }
         }
-        return conn;
+        return requestBuilder.build();
     }
 
     @Override
     public InputStream connect(@NonNull Uri uri, @NonNull ConnectionParameters params)
             throws Exception {
+        Request request = buildRequest(uri, params);
+        mCall = sOkHttpClient.newCall(request);
```

Comment:
mCall is a class level variable here. It's only useful to keep track of it in case of cancellation. We can only track and cancel one call at a time with this implementation, but this was also a limitation with the previous implementation.
This SDK also doesn't make use of the cancel function